src/trainer.py

```diff
@@ -44,7 +44,13 @@ def _get_task_attr(attr_name): return config.get_task_attr(
     for attr in train_opts:
         params[attr] = _get_task_attr(attr)
     for attr in extra_opts:
-        params[attr] = getattr(args, attr)
+        if attr == 'training_data_fraction':
+            if phase == 'pretrain':
+                params[attr] = getattr(args, 'pretrain_data_fraction')
+            else:
+                params[attr] = getattr(args, 'target_train_data_fraction')
+        else:
+             params[attr] = getattr(args, attr)
```

Comment:
nit: `args[attr]` should also work, and strongly prefer `args.pretrain_data_fraction` to using getattr with static arguments.